### ISSUE TITLE
the --help and --version parameters seems to not get parsed correctly because they are default commands 

### ISSUE THREAD

**Issue by OrHayat** [BrianPugh/cyclopts#219]
this is simplified version of the code i wrote

wrote some commands that build some kind of debian package
one parameter was optional version if it wasnt passed i would detect it from github/config file 
when i passed version i wanted to insert the passed version to the debian package instead to skip the detecting step
it seemed that --version flag wasnt working so i wrote this small command to see the generated control file
for debugging and simplified it for this example

```python3
import logging

from cyclopts import App,Parameter
app=App()
from typing import Annotated
@app.command(name="print-control-file")
def print_control(version:Annotated[str|None,Parameter(help="control file version")]):
    x=f"""Package: <package name>
Version:{version}-release
Architecture: amd64
Section: admin
Priority: optional
Depends: python3,lsof
Recommends:
"""
    print(x)
def main():
    logging.basicConfig(level=logging.DEBUG, format="%(name)s - %(levelname)s - %(message)s", stream=sys.stderr)
    app()


````

```bash
 > ./cli.py  print-control-file --version 1.2.3 -h
Usage: cli.py print-control-file [ARGS] [OPTIONS]

╭─ Parameters ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ *  VERSION,--version  control file version [required]                                                                             │
╰─────────────────────────────────────────────────────────────
```

```bash
./cli.py  print-control-file --version 1.2.3
0.0.0
````

```bash
./cli.py  print-control-file --version=1.2.3
Package: <package name>
Version:1.2.3-release
Architecture: amd64
Section: admin
Priority: optional
Depends: python3,lsof
Recommends:
```

i tried to check overriding --help
and i got same behaviour

so  i thought thats bug either in allowing such flags to be registered or by 
parsing the command args

checked 0 parameters commands but those worked so i think it only for default commands that start with --help --version or -h but i might be wrong

for example i tried( and it worked)
```python3
from cyclopts import App,Parameter
app=App()
from typing import Annotated

@app.command(name="--test")
def test():
    print('--test was called')

@app.command(name="print")
def print_test(test:Annotated[str,Parameter(name="--test")]="value"):
    if test:
        print("print_test was called",test)
def main():
    logging.basicConfig(level=logging.DEBUG, format="%(name)s - %(levelname)s - %(message)s", stream=sys.stderr)
    app()


if __name__ == "__main__":
    main()
```
after that i checked with typer to see if/how it behave in such cases.
( it dont have --version  param or -h by default)
so i checked --help param overriding
and typer handled that one correctly
```python3

import typer
from typer import Option
app=typer.Typer(name="test")

@app.command(name="print2")
def print_help_test_2(help_msg:Annotated[str,Option("--help")]="help-message-content"):
    print("test command was called with msg:",help_msg)
    

@app.command(name="print")
def print_help_test(print_help:Annotated[bool,Option("--help")]=False):
    if print_help:
        print("--help was passed")
    else:
        print("--help wasn't passed")

def main():
    app()
if __name__ == "__main__":
    main()

````

```bash
 > ./cli.py print --help
--help was passed

 > ./cli.py print2 --help help-value
```

**Comment by BrianPugh** [comment]
Thanks for opening this issue! To see if I can summarize, the issue is **when a command parameter has the same name as a help/version flag**, correct?

Typically, CLI's handle `--help` in a special manner. Specifically, the `--help` can be provided at any location in the CLI invocation, and it will work (this isn't universally true, but I typically like it when it does).

I'll need to think about how to resolve this, because I definitely think your use-case/issue is valid.

**Comment by OrHayat** [comment]
@BrianPugh yes

**Comment by BrianPugh** [comment]
resolved (enough) in v2.9.8.
